apps/web/src/components/forms/status-page-form.tsx

```diff
@@ -92,12 +93,11 @@ export function StatusPageForm({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [checkUniqueSlug]);
 
-  // FIXME: 🐛 This is causing a bug
-  // useEffect(() => {
-  //   if (!watchSlug) {
-  //     form.setValue("slug", slugify(watchTitle));
-  //   }
-  // }, [watchTitle, form, watchSlug]);
+  useEffect(() => {
+    if (!defaultValues?.title) {
+      form.setValue("slug", slugify(watchTitle));
+    }
+  }, [watchTitle, form, watchSlug, defaultValues?.title]);
```

Comment:
You'll need to remove `watchSlug` from deps, otherwise we won't be able to customize the slug on page creation. Otherwise, lgtm!

```suggestion
  }, [watchTitle, form, defaultValues?.title]);
```